__tests__/warehouse.test.js

```diff
@@ -810,6 +810,42 @@ describe("handle recordId from cloud sources", () => {
     });
   });
 
+  it("should set id based on recordId when sourceCategory is singer-protocol and has source version", () => {
```

Comment:
Having source version check I could not find in this PR. Any version in message.context.sources suffice for this?

Reply:
Yeah, any version would suffice